L1Trigger/TrackFindingTMTT/plugins/TMTrackProducer.cc

```diff
@@ -0,0 +1,380 @@
+#include <L1Trigger/TrackFindingTMTT/plugins/TMTrackProducer.h>
```

Comment:
The CMS rules do allow the option of having a separate header file, providing it is kept in plugins/ . I don't understand the motivation for eliminating the header file? Having separate headers makes it easier to understand what a class does.

Reply:
separate headers primarily serve to separate interface and implementation, which facilitates linking
however, no code should link to plugins, so they rarely need separate headers
having the class definition at the top of the .cc file is just as easy to read

Reply:
We will do this if you insist. However, since the CMS rules allow the option of the separate files, and  the authors find the code easier to read with separate files, could we ask if you'd be OK with this?

Reply:
I have a personal preference for avoiding separate .h files when it's not needed, as well.
But using them doesn't harm particularly, so I see no reason for insisting.